tt_metal/impl/device/mesh_device.hpp

```diff
@@ -72,38 +91,42 @@ class SystemMesh {
 
     // Get the physical device IDs mapped to a MeshDevice
     std::vector<chip_id_t> get_mapped_physical_device_ids(const MeshDeviceConfig &config) const;
+    void register_mesh_device(const std::shared_ptr<MeshDevice> &mesh_device, const std::vector<Device*>& devices);
 
     // Map MeshDevice to physical devices
     std::vector<Device *> map_mesh_device(
         std::shared_ptr<MeshDevice> mesh_device,
-        size_t num_command_queues,
-        size_t l1_small_size,
-        size_t trace_region_size,
+        std::size_t num_command_queues,
+        std::size_t l1_small_size,
+        std::size_t trace_region_size,
         DispatchCoreType dispatch_core_type,
-        const std::pair<size_t, size_t> &offset = {0, 0},
-        const std::vector<chip_id_t> &physical_device_ids = {});
+        const MeshDeviceConfig &config);
 
     // Unmap MeshDevice, releasing the associated physical devices.
-    void unmap_mesh_device(const std::shared_ptr<MeshDevice> &mesh_device);
+    void unmap_mesh_device(const MeshDevice* mesh_device);
+    std::shared_ptr<MeshDevice> get_mesh_device(const std::vector<chip_id_t>& physical_device_ids);
+    Device* get_device(const chip_id_t physical_device_id) const;
 };
 
 class MeshDevice : public std::enable_shared_from_this<MeshDevice> {
```

Comment:
for separate, non pr-review discussion:
We are confusing people using same names for things of different meanings.
Device and MeshDevice
MeshDevice does not look like a device (from api standpoint), and can't be used instead of a device.

Reply:
We can sync offline if you have further follow-ups. The intent is to have MeshDevice have a Device-like contract/interface as we start integrating this as metal-native